interim_results.py

```diff
@@ -0,0 +1,128 @@
+                print(f'Error while sending: {str(e)}')
+                raise
+
+        async def receiver(ws):
+            nonlocal clock_cursor, audio_cursor, transcript_cursor
+            transcript = ''
+            last_word_end = 0.0
+            latest_final_result_processed = False
+            async for msg in ws:
+                msg = json.loads(msg)
+
+                transcript_cursor = msg['start'] + msg['duration']
+
+                if msg['is_final'] and len(msg['channel']['alternatives'][0]['transcript']) > 0:
+                    latest_final_result_processed = False
+                    transcript += ' '
+                    transcript += msg['channel']['alternatives'][0]['transcript']
+                    last_word_end = msg['channel']['alternatives'][0]['words'][-1]['end']
+                    line = f'Clock: {clock_cursor:07.3f}, Audio: {audio_cursor:07.3f}, Transcript: {transcript_cursor:07.3f}; {transcript}'
+                    print(line)
+                    # using end='\r' doesn't work if the line wraps
+                    # this moves the cursor up and overwrites the line regardless of length
+                    # https://stackoverflow.com/a/47170056
+                    print("\033[{}A".format(len(line) // int(terminal_size.columns) + 1), end='')
+
+                elif not msg['is_final']:
+                    if len(msg['channel']['alternatives'][0]['transcript']) > 0:
+                        current_word_begin = msg['channel']['alternatives'][0]['words'][0]['start']
+                        if current_word_begin - last_word_end > SILENCE_INTERVAL:
+                            if len(transcript) > 0:
+                                print(f'Clock: {clock_cursor:07.3f}, Audio: {audio_cursor:07.3f}, Transcript: {transcript_cursor:07.3f}; {transcript}')
+                                beepy.beep(sound=1)
+                                transcript = ''
+                            latest_final_result_processed = True
+                    else:
+                        if transcript_cursor - last_word_end > SILENCE_INTERVAL:
+                            if len(transcript) > 0:
+                                print(f'Clock: {clock_cursor:07.3f}, Audio: {audio_cursor:07.3f}, Transcript: {transcript_cursor:07.3f}; {transcript}')
+                                beepy.beep(sound=1)
+                                transcript = ''
+                            latest_final_result_processed = True
+
+
+        await asyncio.wait([
+            asyncio.ensure_future(microphone()),
+            asyncio.ensure_future(sender(ws)),
+            asyncio.ensure_future(receiver(ws))
+        ])
+
+def parse_args():
+    """ Parses the command-line arguments.
+    """
+    parser = argparse.ArgumentParser(description='Submits data to the real-time streaming endpoint.')
+    parser.add_argument('-k', '--key', required=True, help='YOUR_DEEPGRAM_API_KEY (authorization)')
+    return parser.parse_args()
```

Comment:
should `SILENCE_INTERVAL` also be configurable?

Reply:
oh yeah, I can set it up so it defaults to 2.0 but can be configured via CLI